packages/dds/merge-tree/src/test/client.rollback.spec.ts

```diff
@@ -181,6 +181,59 @@ describe("client.rollback", () => {
             assert(props === undefined || props.foo === undefined);
         }
     });
+    it("Should rollback annotate that later gets split", () => {
+        client.insertTextLocal(0, "abfg");
+        client.annotateRangeLocal(0, 4, { foo: "bar" }, undefined);
+        client.insertTextLocal(1, "cde");
+        client.rollback?.({ type: MergeTreeDeltaType.INSERT }, client.peekPendingSegmentGroups());
+        client.rollback?.({ type: MergeTreeDeltaType.ANNOTATE }, client.peekPendingSegmentGroups());
+
+        assert.equal(client.getText(), "abfg");
+        for (let i = 0; i < 4; i++) {
+            const props = client.getPropertiesAtPosition(i);
+            assert(props === undefined || props.foo === undefined);
```

Comment:
We have to check for either condition because the rollback might unset `foo` in the props, but not remove the props altogether (even though they didn't exist before the annotate)?

Reply:
I think in this case it happens to be undefined but we should treat empty and undefined props the same. There are situations where either can happen and they should be equivalent.